src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php

```diff
@@ -2089,6 +2076,17 @@ private function registerMessengerConfiguration(array $config, ContainerBuilder
             $container->removeDefinition('console.command.messenger_failed_messages_show');
             $container->removeDefinition('console.command.messenger_failed_messages_remove');
         }
+
+        if (false === $config['reset_on_message']) {
+            throw new LogicException('The "framework.messenger.reset_on_message" configuration option can be set to "true" only. To prevent services resetting after each message you can set the "--no-reset" option in "messenger:consume" command.');
+        }
+
+        if (null === $config['reset_on_message']) {
+            trigger_deprecation('symfony/framework-bundle', '5.4', 'Not setting the "framework.messenger.reset_on_message" configuration option is deprecated, it will default to "true" in version 6.0.');
+
+            $container->getDefinition('console.command.messenger_consume_messages')->replaceArgument(5, null);
```

Comment:
I will debug further and create a proper issue/PR later but this service may not exist:

```
Symfony\Component\DependencyInjection\Exception\ServiceNotFoundException: You have requested a non-existent service "console.command.messenger_consume_messages".
```

Reply:
Yep, I've found where it is. It's possible when `messenger` is installed without `console`. I will fix it, but I don't know how to test that fix because of `class_exists(Application::class)` expression.